dhall/src/Dhall/TypeCheck.hs

```diff
@@ -3687,7 +3687,7 @@ prettyTypeMessage (CantProjectByExpression expr) = ErrorMessages {..}
 
 prettyTypeMessage (MissingField k expr0) = ErrorMessages {..}
   where
-    short = "Missing record field"
+    short = "Missing record field: " <> Pretty.pretty k
```

Comment:
Actually I think that this wouldn't backtick-quote "special" labels like `123`. If you don't mind I think this would be worth fixing.